core/src/main/java/org/elasticsearch/index/reindex/BulkByScrollTask.java

```diff
@@ -170,12 +173,9 @@ public WorkerBulkByScrollTaskState getWorkerState() {
 
     @Override
     public void onCancelled() {
-        if (isLeader()) {
-            // The task cancellation task automatically finds children and cancels them, nothing extra to do
-        } else if (isWorker()) {
+        // The task cancellation task automatically finds children and cancels them, nothing extra to do for leaders
```

Comment:
It'd be nice to also mention that tasks that do not yet know if they are leaders or followers also have nothing to do as they can wait.